lib/ansible/modules/cloud/amazon/ec2_lc.py

```diff
@@ -181,169 +221,233 @@ def create_block_device(module, volume):
+        launch_config['IamInstanceProfile'] = instance_profile_name
+
+    if assign_public_ip is not None:
+        launch_config['AssociatePublicIpAddress'] = assign_public_ip
+
+    if user_data is not None:
+        launch_config['UserData'] = user_data
+
+    if ebs_optimized is not None:
+        launch_config['EbsOptimized'] = ebs_optimized
+
+    if len(launch_configs) == 0:
         try:
-            connection.create_launch_configuration(lc)
-            launch_configs = connection.get_all_launch_configurations(names=[name])
+            connection.create_launch_configuration(**launch_config)
+            launch_configs = connection.describe_launch_configurations(LaunchConfigurationNames=[name]).get('LaunchConfigurations')
             changed = True
-        except BotoServerError as e:
-            module.fail_json(msg=str(e))
+            if len(launch_configs) > 0:
+                launch_config = launch_configs[0]
+            else:
+                module.fail_json(msg="Failed to create launch configuration", exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+        except botocore.exceptions.ClientError as e:
+            module.fail_json(msg="Failed to create launch configuration", exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
 
-    result = dict(
-                 ((a[0], a[1]) for a in vars(launch_configs[0]).items()
-                  if a[0] not in ('connection', 'created_time', 'instance_monitoring', 'block_device_mappings'))
-        )
-    result['created_time'] = str(launch_configs[0].created_time)
-    # Looking at boto's launchconfig.py, it looks like this could be a boolean
-    # value or an object with an enabled attribute.  The enabled attribute
-    # could be a boolean or a string representation of a boolean.  Since
-    # I can't test all permutations myself to see if my reading of the code is
-    # correct, have to code this *very* defensively
-    if launch_configs[0].instance_monitoring is True:
-        result['instance_monitoring'] = True
-    else:
-        try:
-            result['instance_monitoring'] = module.boolean(launch_configs[0].instance_monitoring.enabled)
-        except AttributeError:
-            result['instance_monitoring'] = False
-    if launch_configs[0].block_device_mappings is not None:
-        result['block_device_mappings'] = []
-        for bdm in launch_configs[0].block_device_mappings:
-            result['block_device_mappings'].append(dict(device_name=bdm.device_name, virtual_name=bdm.virtual_name))
-            if bdm.ebs is not None:
-                result['block_device_mappings'][-1]['ebs'] = dict(snapshot_id=bdm.ebs.snapshot_id, volume_size=bdm.ebs.volume_size)
+    result = (dict((k, v) for k, v in launch_config.items()
+              if k not in ['Connection', 'CreatedTime', 'InstanceMonitoring', 'BlockDeviceMappings']))
+
+    result['CreatedTime'] = str(launch_config.get('CreatedTime'))
```

Comment:
You may want to import to_text() from ansible.module_utils._text and use that instead of str() to be sure of python3 compatibility. (But having trouble testing that since boto3 isn't returning the created time for me...)